src/main/java/com/google/maps/RadarSearchRequest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.google.maps;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.maps.errors.ApiException;
+import com.google.maps.internal.ApiConfig;
+import com.google.maps.internal.ApiResponse;
+import com.google.maps.model.*;
+
+public class RadarSearchRequest
+    extends PendingResultBase<PlacesSearchResponse, RadarSearchRequest, RadarSearchRequest.Response> {
+
+  static final ApiConfig API_CONFIG = new ApiConfig("/maps/api/place/radarsearch/json")
+      .fieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES);
+
+  protected RadarSearchRequest(GeoApiContext context) {
+    super(context, API_CONFIG, Response.class);
+  }
+
+  /**
+   * location is the latitude/longitude around which to retrieve place information.
+   */
+  public RadarSearchRequest location(LatLng location) {
+    return param("location", location);
+  }
+
+  /**
+   * radius defines the distance (in meters) within which to bias place results.
```

Comment:
The comment in NearbySearch gives a big warning about 50,000 but doesn't validate the argument. Here, you validate the argument but not give a warning in the comment.